spark/core/src/main/scala/org/elasticsearch/spark/serialization/ScalaValueWriter.scala

```diff
@@ -61,7 +61,14 @@ class ScalaValueWriter(writeUnknownTypes: Boolean = false) extends JdkValueWrite
         generator.writeBeginObject()
         for ((k, v) <- m) {
           if (shouldKeep(parentField, k.toString)) {
-            if (v != None && v!= null && v != () || hasWriteNullValues) {
+            val nonEmptyValue = Option(v) match {
```

Comment:
```suggestion
            val hasValue = Option(v) match {
```

Can we swap the variable name? `nonEmptyValue=false` reads weird in my head